File: frontend/src/types/solar.ts
export interface SunPosition {
  time: string; // ISO datetime
  azimuth: number; // degrees from north, clockwise
  elevation: number; // degrees above horizon
  is_daytime: boolean;
}

export interface SolarDayData {
  date: string;
  sunrise: string;
  solar_noon: string;
  sunset: string;
  day_length_hours: number;
  hourly_path: SunPosition[];
}

export interface SolarMonthData {
  month: number;
  avg_irradiance_kwh: number;
  avg_peak_sun_hours: number;
  avg_day_length_hours: number;
}

export interface SolarSeasonalData {
  location: { lat: number; lon: number };
  monthly: SolarMonthData[];
  summer_solstice: SolarDayData;
  winter_solstice: SolarDayData;
}

File: backend/app/schemas/solar.py
from datetime import datetime

from pydantic import BaseModel


class SunPositionOut(BaseModel):
    time: str           # ISO 8601
    azimuth: float      # degrees clockwise from north
    elevation: float    # degrees above horizon
    is_daytime: bool


class SolarDayOut(BaseModel):
    date: str
    sunrise: str
    solar_noon: str
    sunset: str
    day_length_hours: float
    max_elevation_deg: float
    hourly_path: list[SunPositionOut]


class SolarMonthOut(BaseModel):
    month: int
    avg_irradiance_kwh: float
    avg_peak_sun_hours: float
    avg_day_length_hours: float


class SolarSeasonalOut(BaseModel):
    lat: float
    lon: float
    monthly: list[SolarMonthOut]
    summer_solstice: SolarDayOut
    winter_solstice: SolarDayOut
